src/Adapter/FilesystemItems.php

```diff
@@ -0,0 +1,209 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Libero\ContentApiBundle\Adapter;
+
+use CallbackFilterIterator;
+use EmptyIterator;
+use FilesystemIterator;
+use FluentDOM\Utility\Iterators\MapIterator;
```

Comment:
Will be slightly annoying to have this as a dependency when this is extracted into a separate library. Possibly best to duplicate at that point (unless there's a better way?).

Reply:
The iterator or the whole library?

Reply:
`MapIterator` sounds like something that should come from a standard library, and since it isn't needs to be duplicate or extracted into its own. Strange that `FluentDOM` doesn't take it from somewhere else like https://github.com/lstrojny/functional-php and similar

Reply:
That and [`nikic/iter`](https://github.com/nikic/iter) could make it a lot cleaner. Trying to avoid too many dependencies though.